kolibri/core/discovery/utils/network/client.py

```diff
@@ -115,13 +115,13 @@ def post(self, path, **kwargs):
 
     def request(self, method, path, **kwargs):
         response = None
-        if "timeout" not in kwargs:
-            kwargs.update(timeout=self.timeout)
+
+        timeout = kwargs.pop("timeout", self.timeout)
 
         url = join_url(self.base_url, path)
         try:
             with super(NetworkClient, self).request(
-                method, url, stream=True, **kwargs
+                method, url, timeout=timeout, stream=True, params=kwargs
```

Comment:
This is very likely a breaking change. Not all requests use `params`, and seeing as this overrides the base `request` method which the underlying methods call (`get`, `post`, etc), this change leads to it not properly passing all parameters to the super method.

Reply:
Ahh right on - I'll give this all a closer look and might ping you on Slack for a bit more clarifications around this - even if we revert this here it seems like some changes are needed around here in any case and I can make a follow-up issue